### ISSUE TITLE
[R] Implement halffloat -> R conversion

### ISSUE THREAD

**Issue by asfimport** [apache/arrow#23991]
Followup to ARROW-3783

**Reporter**: [Neal Richardson](https://issues.apache.org/jira/browse/ARROW-7753) / @nealrichardson
#### Related issues:
- [[C++] Cast to/from halffloat not implemented](https://github.com/apache/arrow/issues/20213) (is blocked by)

<sub>**Note**: *This issue was originally created as [ARROW-7753](https://issues.apache.org/jira/browse/ARROW-7753). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

**Comment by vkhodygo** [comment]
Is it still actual? I noticed I can't read halffloat values correctly when I load a `parquet` file in R. Not sure if I need to open a new issue.

**Comment by vkhodygo** [comment]
@nealrichardson Is it completely abandoned?

**Comment by nealrichardson** [comment]
Looking at the linked #20213, the C++ library appears to have some float16 support now, so this should be doable. I don't personally intend to pick this up but someone could.
